stackit/internal/services/ske/project/resource.go

```diff
@@ -182,7 +183,24 @@ func (r *projectResource) Delete(ctx context.Context, req resource.DeleteRequest
 	ctx = tflog.SetField(ctx, "project_id", projectId)
 
 	c := r.client
-	_, err := c.DisableService(ctx, projectId).Execute()
+
+	clusters, err := c.ListClusters(ctx, projectId).Execute()
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error getting the list of clusters", fmt.Sprintf("Calling API: %v", err))
+		return
+	}
+
+	clusterCount := len(*clusters.Items)
+	if clusterCount > 0 {
+		if clusterCount == 1 {
```

Comment:
I think we can just have on error message saying "You still have clusters in your project. Please delete them before deleting the project."

No need to specify the number of clusters and then we make this error handling simpler. 
And I think specifying that the user needs to delete the clusters is more clear than just calling it "resources"